packages/xo-server/src/api/vm.mjs

```diff
@@ -774,6 +774,29 @@ set.resolve = {
 
 // -------------------------------------------------------------------
 
+export const setAndRestart = defer(async function ($defer, params) {
+  const vm = params.VM
+  const force = extract(params, 'force')
+
+  await stop.bind(this)({ vm, force })
+
+  $defer(start.bind(this), { vm, force })
+
+  return set.bind(this)(params)
+})
+
+setAndRestart.params = {
+  // Restart options
+  force: { type: 'boolean', optional: true },
+
+  // Set params
+  ...set.params,
+}
+
+setAndRestart.resolve = set.resolve
```

Comment:
This is not necessary and could be implemented on the client (xo-web) side.

Reply:
If the target VM is between the XO and the user,  it won't restart, the ticket example is with a pfsense VM that is probably  in this case, but also any running reverse proxy

Reply:
I agree for the reverse proxy.

Regarding pfSense, I don't think temporary shutting down the DHCP machine will prevent this action for working, and the ticket description, the user is saying that pfSense is also providing the IP address to the host, in that case, that would also break the connection between XO and the host.

Anyway, the last decision in the issue was to do it on the UI side, if it's necessary to do it differently, we should discuss about it and test the problematic use case before and after the fix.

I propose we implement the minimal solution (proposed in the ticket), and if that still does not work for the user, we'll try another one.

@b-Nollet  @fbeauchamp What do you think?

Reply:
I think it will be more robust on the server side, by far, with the UI handling the fallback to the reboot and set 
Any page reloading will let the VM in an incorrect state for example